src/Sentry/Sentry.csproj

```diff
@@ -23,21 +23,24 @@
   <Import Project="Platforms\iOS\Sentry.iOS.props" Condition="'$(TargetPlatformIdentifier)' == 'ios' Or '$(TargetPlatformIdentifier)' == 'maccatalyst'" />
   <Import Project="Platforms\Native\Sentry.Native.props" Condition="('$(TargetFramework)' == 'net7.0' Or '$(TargetFramework)' == 'net8.0') And ($([MSBuild]::IsOSPlatform('Linux')) Or $([MSBuild]::IsOSPlatform('Windows')))" />
 
-  <!--
-    Ben.Demystifier is compiled directly into Sentry.
-    Note: It uses Microsoft.Bcl.AsyncInterfaces, which we get transitively from System.Text.Json.
-  -->
-  <ItemGroup>
-    <Compile Include="..\..\modules\Ben.Demystifier\src\**\*.cs">
-      <Link>%(RecursiveDir)\%(Filename)%(Extension)</Link>
-    </Compile>
-    <Compile Remove="..\..\modules\Ben.Demystifier\**\obj\**" />
-  </ItemGroup>
+  <Target Name="IncludeBenDemystifier" BeforeTargets="BeforeBuild" Condition="'$(PublishAot)' != 'true'">
```

Comment:
`PublishAot` here in a library package is confusing to me because we're not actually publishing Sentry as a native AOT library. Or at least I don't think we need to? 🤔 

Maybe we could define a condition based on the solution name instead?